dsp-repository/v1/dsp-repository.shacl.ttl

```diff
@@ -26,32 +26,739 @@
+    sh:targetClass dsp-repo:Organization ;
+    sh:property [
+        sh:path dsp-repo:hasOrganizationIdentifier ;
+        sh:name "Identifier" ;
+        sh:description "Identifier from https://www.grid.ac/"@en ;
+        sh:or ( [ sh:datatype xsd:string ] [ sh:nodeKind sh:IRI ] ) ;
+        sh:maxCount 1 ;
+    ] ;
+    sh:property [
+        sh:path dsp-repo:hasOrganizationName ;
+        sh:name "Legal name" ;
+        sh:description "Legal name"@en ;
+        sh:datatype xsd:string ;
+        sh:minCount 1 ;
+    ] ;
+    sh:property [
+        sh:path dsp-repo:hasOrganizationEmail ;
+        sh:name "Organization e-mail" ;
+        sh:description "E-mail address"@en ;
+        sh:datatype xsd:string ;
+        sh:maxCount 1 ;
+    ] ;
+    sh:property [
+        sh:path dsp-repo:hasOrganizationAddress ;
+        sh:name "Address" ;
+        sh:description "Postal address"@en ;
+        sh:datatype xsd:string ;
+        sh:maxCount 1 ;
+    ] ;
+    sh:property [
+        sh:path dsp-repo:hasOrganizationURL ;
+        sh:name "URL" ;
+        sh:description "URL"@en ;
+        sh:datatype xsd:string ;
+        sh:maxCount 1 ;
+    ] ;
+    sh:closed true ;
+	sh:ignoredProperties ( rdf:type ) .
+
+### --- Class definition of dsp-repo:Grant --- 
+# http://ns.dasch.swiss/repository#Grant
+dsp-repo:Grant 
+    a rdfs:Class ;
+    rdfs:comment "A grant"@en ;
+    rdfs:label "dsp-repo:Grant"@en ;
+    rdfs:subClassOf schema:Grant . 
+
+### --- Properties definition of dsp-repo:Grant ---
+# http://ns.dasch.swiss/repository#hasName
+dsp-repo:hasName
+    a rdf:Property ;
+    rdfs:subPropertyOf schema:name .
+
+# http://ns.dasch.swiss/repository#hasGrantURL
+dsp-repo:hasGrantURL
+    a rdf:Property ;
+    rdfs:subPropertyOf schema:url .
+
+# http://ns.dasch.swiss/repository#hasGrantNumber
+dsp-repo:hasGrantNumber
+    a rdf:Property .
+
+# http://ns.dasch.swiss/repository#hasFunder
+dsp-repo:hasFunder
+    a rdf:Property ;
+     rdfs:subPropertyOf schema:funder .
+
+### --- Shape of dsp-repo:Grant ---
+dsp-repo:GrantShape
+    a sh:NodeShape ;
+    sh:targetClass dsp-repo:Grant ;
+    sh:property [
+        sh:path dsp-repo:hasName ;
+        sh:name "Name" ;
+        sh:description "Name of the grant [Ambizione, etc.]"@en ;
+        sh:datatype xsd:string ;
+        sh:maxCount 1 ;
+    ] ;
+    sh:property [
+        sh:path dsp-repo:hasGrantURL ;
+        sh:name "URL" ;
+        sh:description "URL of the grant"@en ;
+        sh:datatype xsd:string ;
+        sh:maxCount 1 ;
+    ] ; 
+     sh:property [
+        sh:path dsp-repo:hasGrantNumber ;
+        sh:name "Number" ;
+        sh:description "Number of the grant"@en ;
+        sh:datatype xsd:string ;
+        sh:maxCount 1 ;
+    ] ; 
+     sh:property [
+        sh:path dsp-repo:hasFunder ;
+        sh:name "Funder" ;
+        sh:description "Funder of the grant"@en ;
+        sh:or ( [ sh:datatype xsd:string ] [ sh:nodeKind sh:IRI ] ) ;
+        sh:maxCount 1 ;
+    ] ; 
```

Comment:
That works in my opinion, but I don't like that we now define `hasFunder` twice (here and line 180ff.), both practically identical, but not quite.

Instead I'd suggest defining something like this:
```ttl
dsp-repo:funderPropertyShape
    a sh:propertyShape;
    sh:path dsp-repo:hasFunder
    ... etc.
```
(or something like that)  
and then, both in `dsp-repo:projectShape` and `dsp-repo:grantShape`, we could simply call:
```ttl
    sh:property dsp-repo:funder-property-shape ; 
```

If that seems good to you, I'll make these changes.

Reply:
Has been solved. It is no longer twice in it.

Reply:
Solved. It is not twice in it anymore.